src/Microsoft.OData.Core/Metadata/EdmLibraryExtensions.cs

```diff
@@ -1826,23 +1930,33 @@ private static bool ParametersSatisfyFunction(IEdmOperation operation, IList<str
                 parametersToMatch = parametersToMatch.Skip(1);
             }
 
-            List<IEdmOperationParameter> functionParameters = parametersToMatch.ToList();
-
             // if any required parameters are missing, don't consider it a match.
-            if (functionParameters.Where(
-                p => !(p is IEdmOptionalParameter)).Any(
-                    p => parameterNameList.All(
-                        k => !string.Equals(k, p.Name, caseInsensitive ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal))))
+            BitArray parameterExists = new BitArray(parameterNameList.Count);
+            foreach (var functionParameter in parametersToMatch)
             {
-                return false;
+                bool matched = false;
+                for (int j = 0; j < parameterNameList.Count; j++)
+                {
+                    if (string.Equals(parameterNameList[j], functionParameter.Name, caseInsensitive ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal))
+                    {
+                        matched = true;
+                        parameterExists[j] = true;

```

Comment:
break;

Reply:
It doesn't look like you added the break; is there a reason you have to continue looping through the remaining parameter names once you find one that matches?

---
In reply to: [370884636](https://github.com/OData/odata.net/pull/1602#discussion_r370884636) [](ancestors = 370884636)

Reply:
Discussed. 

---
In reply to: [371602187](https://github.com/OData/odata.net/pull/1602#discussion_r371602187) [](ancestors = 371602187,370884636)

Reply:
More specifically; this matches the current (flawed) behavior of flagging *all* case-insensitive matches. Kanish to open a separate issue to track fixing this existing logic.

---
In reply to: [371616803](https://github.com/OData/odata.net/pull/1602#discussion_r371616803) [](ancestors = 371616803,371602187,370884636)

Reply:
[https://github.com/OData/odata.net/issues/1649](https://github.com/OData/odata.net/issues/1649)

---
In reply to: [371618735](https://github.com/OData/odata.net/pull/1602#discussion_r371618735) [](ancestors = 371618735,371616803,371602187,370884636)